llvm/tools/llvm-profdata/llvm-profdata.cpp

```diff
@@ -2406,6 +2407,14 @@ static void traverseAllValueSites(const InstrProfRecord &Func, uint32_t VK,
   }
 }
 
+namespace {
+struct InstrProfilePerFuncOptions {
```

Comment:
Will it be limited to InstrProfile? Perhaps just name it "PerFuncOptions"?

Reply:
Showing indirect call targets seems implicit currently; `showSampleProfile` doesn't take an option to suppress indirect call targets. 

A minor thing is, functions in sample profiles doesn't have mem-op size; so passing 'PerFuncOptions' in `showSampleProfile` static function would require an option invalidation 

```
if (PerFuncOptions.ShowMemOPSizes)
  emit error with message "mem-op size is not supported in sample profiles"
```


The struct is in anonymous namespace which makes it easier for renames if necessary.